index.bs

```diff
@@ -1715,3 +1705,220 @@ When [=a browsing context is discarded=], run the following steps:
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |realm| be the browsing context with id |realm id|.
+
+ 1. Return [=success=] with data |realm|
+
+Issue: This has the wrong error code
+
+</div>
+
+<div algorithm>
+To <dfn>get the realm info</dfn> given |environment settings|:
+
+ 1. Let |realm| be |environment settings|' [=realm execution context=]'s Realm component.
+
+ 1. Let |realm id| be the [=realm id=] for |realm|.
+
+ 1. Run the steps under the first matching condition:
+
+   <dl>
+     <dt>The [=Realm/global object=] specified by |environment settings| is a [=Window=] object
+     <dd>
+      1. If [=Realm/global object=] specified by |environment settings| has a [=Document=]
+         that is not yet fully active, return null.
+
+      1. Let |type| be "<code>window</code>".
+
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{DedicatedWorkerGlobalScope}} object
+     <dd>
+       1. Let |type| be "<code>worker</code>".
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{SharedWorkerGlobalScope}} object
+
+    <dd>
+       1. Let |type| be "<code>sharedworker</code>".
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{ServiceWorkerGlobalScope}} object
+
+    <dd>
+       1. Let |type| be "<code>serviceworker</code>".
+
+    <dt>The [=Realm/global object=] specified by |environment settings| is a {{WorkletGlobalScope}} object
+    <dd>
+       1. Let |type| be "<code>worklet</code>".
+
+    <dt>Otherwise:
+    <dd>
+      1. Let |type| be an implementation-defined string, consisting of a prefix,
+         followed by a colon ("<code>:</code>") followed by a type name.
+   </dl>
+
+ 1. Let |origin| be the [=serialization of an origin=] given |environment settings|'s |origin|.
+
+ 1. Let |realm info| be a map matching the <code>RealmInfo</code> production,
+    with the <code>realm</code> field set to |realm id|, the <code>type</code>
+    field set to |type| and the <code>origin</code> field set to |origin|.
+
+ 1. Return |realm info|
+
+</div>
+
+### Commands ### {#module-script-commands}
+
+#### getRealms ####  {#command-script-getrealms}
+
+The <dfn export for=commands>getRealms</dfn> command returns a list of all
+realms, optionally filtered to realms of a specific type, or to the realm
+associated with the document currently loaded in a specified browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      GetRealmsCommand = {
+        method: "script.getRealms",
+        params: GetRealmsParameters
+      }
+
+      GetRealmsParameters = {
+        ?context: BrowsingContext,
+        ?type: RealmType,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        GetRealmsResult = {
+          realms: [*RealmInfo]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for script.getRealms">
```

Comment:
I haven't reviewed this in detail, but wonder if it will change in light of the discussion of browsing context trees in https://github.com/w3c/webdriver-bidi/pull/62?

@sadym-chromium do you know what represents a JavaScript realm in CDP, and how they're discovered? Is it a `Runtime.something`, a `Target.something`, or something else? I'm not sure what to look for to see if this seems similar to CDP or not :)

Reply:
These are "ExecutionContexts" in CDP: https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#event-executionContextCreated

You get a batch of `Runtime.executionContextCreated` events when the `Runtime` domain is first enabled. I don't recall the exact details but I believe the [ExecutionContextDescription](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-ExecutionContextDescription) has the necessary info to associate it with a Frame.

Reply:
I'm not sure quite what kind of change you're thinking of, so I haven't changed anything yet.

Reply:
Thanks @bwalderman! I don't know anything that needs to be changed at this point.